medical_emr/models/medical_patient_disease.py

```diff
@@ -21,74 +21,82 @@
-            ('da', 'Drug Allergy'),
-            ('fa', 'Food Allergy'),
-            ('ma', 'Misc Allergy'),
-            ('mc', 'Misc Contraindication'),
-        ], string='Allergy type', select=True, sort=False),
-        'patient_id': fields.many2one('medical.patient', string='Patient', ),
-        'age': fields.integer(
-            string='Age when diagnosed',
-            help='Patient age at the moment of the diagnosis.'
-            ' Can be estimative'
-        ),
-        'date_start_treatment': fields.date(
-            string='Start',
-            help='Start of treatment date'
-        ),
-        'short_comment': fields.char(
-            size=256, string='Remarks',
-            help='Brief, one-line remark of the disease. Longer description'
-            ' will go on the Extra info field'
-        ),
-    }
+    @api.one
+    @api.depends('short_comment', 'pathology_id', 'pathology_id.name')
+    def _compute_name(self):
+        name = self.pathology_id.name
+        if self.short_comment:
+            name = '%s - %s' % (name, self.short_comment)
+        self.name = name
+
+    @api.one
+    @api.depends('active')
+    def _compute_expire_date(self):
+        if self.active:
+            self.expire_date = False
+        else:
+            self.expire_date = fields.Datetime.now()
+
+    @api.one
+    def action_invalidate(self):
+        self.active = False
+
+    @api.one
+    def action_revalidate(self):
+        self.active = True
+
+    name = fields.Char(string='Name', compute='_compute_name', store=True)
+    treatment_description = fields.Char(string='Treatment Description')
+    expire_date = fields.Datetime(
+        string='Expire Date', compute='_compute_expire_date', store=True)
+    short_comment = fields.Char(string='Short Comment')
+    pathology_id = fields.Many2one(
+        comodel_name='medical.pathology', string='Pathology', select=True,
+        required=True)
+    physician_id = fields.Many2one(
+        comodel_name='medical.physician', string='Physician', select=True)
+    pcs_code = fields.Many2one(
+        comodel_name='medical.procedure', string='Procedure code', select=True)
+    patient_id = fields.Many2one(
+        comodel_name='medical.patient', string='Patient', required=True,
+        select=True)
+    disease_severity = fields.Selection([
+        ('1_mi', 'Mild'),
+        ('2_mo', 'Moderate'),
+        ('3_sv', 'Severe')
+    ], string='Severity')
+    state = fields.Selection([
+        ('a', 'Acute'),
+        ('c', 'Chronic'),
+        ('u', 'Unchanged'),
+        ('h', 'Healed'),
+        ('i', 'Improving'),
+        ('w', 'Worsening'),
+    ], string='Status of the disease')
+    allergy_type = fields.Selection([
+        ('da', 'Drug Allergy'),
+        ('fa', 'Food Allergy'),
+        ('ma', 'Misc Allergy'),
+        ('mc', 'Misc Contraindication'),
+    ], string='Allergy type')
+    weeks_of_pregnancy = fields.Integer(
+        string='Contracted in pregnancy week #')
```

Comment:
help='Week number of pregnancy when disease contracted'
string='Pregnancy Week #'